packages/core/react/src/WalletProvider.tsx

```diff
@@ -132,10 +132,9 @@ export const WalletProvider: FC<WalletProviderProps> = ({
     const select = useCallback(
         async (walletName: WalletName | null) => {
             if (name === walletName) return;
-            if (adapter) await adapter.disconnect();
             setName(walletName);
         },
-        [name, adapter, setName]
+        [name]
```

Comment:
Is this function needed anymore since `setName` already bails if newName === oldName?

Reply:
Nope! The only weird part is that if you hand people the state setter directly, then instead of a method with the signature `(walletName: WalletName) => void` we'd be handing them a method with the signature `(walletName: WalletName) => void | ((prevWalletName: WalletName) => WalletName) => void)` which is maybe not what we want.

Reply:
Well, in this case I think the state setter is https://github.com/solana-labs/wallet-adapter/blob/422d77d84bdd12dd8a999e8c6c3118128863c2e4/packages/core/react/src/useLocalStorage.ts#L15-L16 so it should be equivalent?

FWIW I tried making this change but got an error from the new tests.

Reply:
I think that #222 will fix this, but I haven't tested that out yet!